docs/user/README.md

```diff
@@ -0,0 +1,100 @@
+# Aeneas user documentation - developer docs
+
+## Architecture
+
+This is a separate `flake.nix` setup from the root repository's one, but it relies on Aeneas' Lean library from this repository.
+
+We pull Lean 4's upstream flake to build Lean in Nix, use some forked Alectryon-related tooling to assemble a mdBook HTML output.
+
+### How to add new files?
+
+If these files are not supposed to be inked by Alectryon (i.e. display interactive goals in the browser), they can be extended following [the vanilla mdBook documentation](https://rust-lang.github.io/mdBook/guide/creating.html) in the `src/` directory.
+
+If these files are supposed to be inked by Alectryon:
+
+- they need to appear in `src/lean/basics` as `xyz.lean` and `xyz.lean.md`
+- the Markdown file version should be included in the `SUMMARY.md` file.
+- the contents of `xyz.lean.md` by default should be the same as the others one, i.e. a Lean code block containing `{{#include xyz.lean}}`
+
+### How to add a new directory containing Lean files?
+
+This requires a change in the `flake.nix`, find the place where the `literate` package is defined:
+
+```nix
+      literate = buildLeanPackage {
+        name = "literate";
+        deps = [ aeneas-base ];
+        src = ./.;
+        roots = [
+          {
+            mod = "src/lean/basics";
+            glob = "submodules";
+          }
+        ];
+      };
+```
+
+Add a new `root` as it follows, assuming your new module is `src/lean/advanced`:
+
+```nix
+      literate = buildLeanPackage {
+        name = "literate";
+        deps = [ aeneas-base ];
+        src = ./.;
+        roots = [
+          {
+            mod = "src/lean/basics";
+            glob = "submodules";
+          }
+          {
+            mod = "src/lean/advanced";
+            glob = "submodules";
+         }
+        ];
+      };
+```
+
+### How to add a new Lean dependency to the project?
+
+Each `buildLeanPackage` can take a list of `deps`.
+
+Two cases:
+
+- either, this is a local dependency and this can be built like the `Base` (Aeneas standard library) one.
+- either, this is a remote dependency and this is often added to a Lakefile, this can be built like `Mathlib` and the others.
+
+## How to build?
+
+```console
+$ nix build .#docs
+```
+
+## How to deploy?
+
+The `result/` symlink is a HTML webroot, this can be deployed as a static website.
+
+## My build is failing due to a compilation error but it works outside of Nix
+
+This is highly likely due to a mismatch between the dependencies in the Nix setup and outside.
+
+The Nix setup reads the `lakefile-manifest.json`, but this lockfile is insufficient to lock purely the dependencies (NAR SHA256 serialization are missing).
+
+To overcome this, they are specified manually in `flake.nix`, but each time the Aeneas standard library updates their dependencies, the `flake.nix` needs to have all its hash rewritten.
```

Comment:
I don't fully understand what we are supposed to do when updating the dependencies, then

Reply:
I expanded much more the documentation, let me know how you feel about it now